remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/ProtobufMetadataManagerImpl.java

```diff
@@ -95,16 +95,24 @@ void start() {
             getProtobufMetadataCacheConfig(globalConfiguration).build(),
             EnumSet.of(InternalCacheRegistry.Flag.USER, InternalCacheRegistry.Flag.PROTECTED, InternalCacheRegistry.Flag.PERSISTENT));
 
-      processSerializationContextInitializer(ServiceFinder.load(SerializationContextInitializer.class, globalConfiguration.classLoader()));
-      processSerializationContextInitializer(globalConfiguration.serialization().contextInitializers());
+      Collection<SerializationContextInitializer> initializers = globalConfiguration.serialization().contextInitializers();
+      if (initializers == null || initializers.isEmpty()) {
+         initializers = ServiceFinder.load(SerializationContextInitializer.class, globalConfiguration.classLoader());
+      }
+      processSerializationContextInitializer(initializers);
       processProtostreamSerializationContextInitializers(globalConfiguration.classLoader());
    }
 
+   /**
+    * @deprecated Since 10.
+    */
+   @Deprecated
    private void processProtostreamSerializationContextInitializers(ClassLoader classLoader) {
       Collection<ProtostreamSerializationContextInitializer> initializers =
             ServiceFinder.load(ProtostreamSerializationContextInitializer.class, classLoader);
 
       for (ProtostreamSerializationContextInitializer psci : initializers) {
+         log.debugf("Registering ProtostreamSerializationContextInitializer %s", psci.getClass().getName());
```

Comment:
Should the logs be info?

Reply:
That was my feeling also in the beginning. I can make them INFO and i18n them, unless we think this is too much noise in the log during boot.

Reply:
+1 for Info and i18n. I think the benefit of easy debugging will outweigh the damage of a little noise.

Reply:
done